internal/pkg/deploy/deploy.go

```diff
@@ -12,3 +12,41 @@ type CreateEnvironmentInput struct {
 	PublicLoadBalancer       bool   // Whether or not this environment should contain a shared public load balancer between applications.
 	ToolsAccountPrincipalARN string // The Principal ARN of the tools account.
 }
+
+// CreatePipelineInput represents the fields required to deploy a pipeline.
+type CreatePipelineInput struct {
+	// Name of the project this pipeline belongs to
+	ProjectName string
+	// Name of the pipeline
+	Name   string
+	Source *Source
+	Stages []PipelineStage
+}
+
+// Source defines the source of the artifacts to be built and deployed.
+type Source struct {
+	ProviderName string
+	Properties   map[string]interface{}
```

Comment:
Having `interface{}` as the value seems odd, is it possible to have these more well defined? How will calling clients know what to put in here?

Reply:
The main reason I chose to do it this way is because I don't want a lot of optional fields when the Provider is different. For example, the `Properties` for GitHub source provider will most likely will be different from a S3 provider